include/phpmailer/class.phpmailer.php

```diff
@@ -869,6 +869,7 @@ public function addReplyTo($address, $name = '')
     protected function addOrEnqueueAnAddress($kind, $address, $name)
     {
         $address = trim($address);
+        $address = html_entity_decode($address, ENT_QUOTES);
```

Comment:
This should be fine for now but ideally this is something I would want to handle in a similar way to #7922. Note that this should be thoroughly tested before being merged.